src/ivq.tsx

```diff
@@ -416,13 +421,23 @@ export class Ivq extends KalturaPlayer.core.BasePlugin {
       })
       .finally(() => {
         this._resolveQuizDataPromise();
-        if (this._dataManager.quizData?.showWelcomePage && !this._player.config.playback.autoplay) {
-          this._player.pause();
-          this._showWelcomeScreen();
-        }
       });
   }
 
+  private _manageWelcomeScreen = (retake = false) => {
+    if (this._dataManager.quizData?.showWelcomePage) {
+      if (this._player.config.playback.autoplay || retake) {
+        this.eventManager.listenOnce(this.player, this.player.Event.PLAY, () => {
+          this._player.pause();
+          this._showWelcomeScreen();
+        });
+      } else {
+        // before first play event duration is NaN
+        this._showWelcomeScreen(Number.isNaN(this._player.duration));
```

Comment:
Better we add small function isFirstPlay() and use it

Reply:
make sense